lib/elixir/lib/integer.ex

```diff
@@ -400,17 +400,23 @@ defmodule Integer do
       iex> Integer.gcd(10, 0)
       10
 
+      iex> Integer.gcd(7, 7)
+      7
+
       iex> Integer.gcd(0, 0)
       0
 
   """
   @spec gcd(integer, integer) :: pos_integer
+  @spec gcd(int, int) :: int when int: pos_integer
+  @spec gcd(0, 0) :: 0
```

Comment:
Exactly, the proof of the existence of the GCD of two integers needs to assume they are both non-null. But then, you have an artificial definition that does not follow the general property for practical reasons. That post... it is all ad-hoc reasonings and by no means a proof in my book, the GCD is not defined for (0, 0), period. So you give it some value that you find in practice makes things smoother, Wikipedia has a reference for example to some paper.

This is similar to excluding 1 from prime numbers. Is it divisible by itself? Yes. Is it divisible by 1? Yes. So it satisfies the criteria! Well, practice suggests that 1 is a pain in the ass doing number theory because it makes you introduce weird exceptions in the theorems. For starters if you want unicity of decomposition of the factorization in prime numbers, you need to exclude 1, or say unicity except for the prime 1, which is not elegant.

In the end, these definitions are human constructions and there is a social consensus behind them, so mathematicians decided with time that it was better to exclude 1 and done. The exception is inserted in the definition, and the theorems flow better.

(Disclaimer, I have a background in math.)